test/extended/build_docker_test.go

```diff
@@ -0,0 +1,61 @@
+// +build extended
+
+package extended
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	"github.com/openshift/origin/test/util"
+)
+
+func init() {
+	util.RequireDocker()
+	util.RequireServer()
+}
+
+func TestBuildDocker(t *testing.T) {
+	namespace := util.RandomNamespace("contextdir")
+	fmt.Printf("Using '%s' namespace\n", namespace)
+
+	build := util.GetBuildFixture("fixtures/docker-build.json")
+	client, _ := util.GetClusterAdminClient(util.KubeConfigPath())
+
+	repo := util.CreateSampleImageRepository(namespace)
+	if repo == nil {
+		t.Fatal("Failed to create ImageRepository")
+	}
+
+	// TODO: Tweak the selector to match the build name
+	watcher, err := client.Builds(namespace).Watch(labels.Everything(), labels.Everything(), "0")
+	if err != nil {
+		t.Fatalf("Failed to create watcher: %v", err)
+	}
+	defer watcher.Stop()
+
+	newBuild, err := client.Builds(namespace).Create(build)
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+
+	for event := range watcher.ResultChan() {
+		build, ok := event.Object.(*buildapi.Build)
+		if !ok {
+			t.Fatalf("cannot convert input to Build")
+		}
+
+		// Iterate over watcher's results and search for
+		// the build we just started. Also make sure that
+		// the build is running, complete, or has failed
+		if build.Name == newBuild.Name {
+			switch build.Status {
+			case buildapi.BuildStatusFailed, buildapi.BuildStatusError:
```

Comment:
I'd add `BuildStatusCancelled` here as well, as it's also an end status of a build.